### ISSUE TITLE
Builds from Unity freeze in HMD

### ISSUE THREAD

**Issue by amiga** [OSVR/OSVR-Unity#209]
I am exporting a working scene from unity, but when I run it it appears on the headset then just freezes (on screen if I have direct mode ticked in my vr cams in project then I can still see the app running and even the head tracking working, but in the headset just a frozen still). I am having to use core snapshot 0.6.1320 because if I go to the newer ones (tried many) I always get back to cannot create render manager. Finding the right combination of unity, osvr core seems almost impossible compared to back when it used to just work!..

**Comment by amiga** [comment]
I dont know why but this has stopped happening, I could use lower res than hd to stop it but now it isnt happening at all.

**Comment by nanospork** [comment]
Just wanted to voice that, after a fresh Windows install, I had this happen too!

Started up a build in Direct Mode and the screen was frozen. Dropped the resolution and it works fine. Went back up and it started up frozen again.

Will report back if it also magically goes away after some time.

**Comment by amiga** [comment]
yes I still observe this in "good" or higher quality around 50% of the time.

**Comment by nanospork** [comment]
Yeah, I'm running into the issue again, in both Direct and Extended mode, even when setting the app at a lower resolution. Happens even with the new package build provided in #208.

**Comment by nanospork** [comment]
Playing around with the quality settings, it looks like it may be related to V-sync. 

@amiga Can you try disabling V-sync to see if that gets rid of the problem for you as well?  (Edit->Project Settings->Quality; go through each of the Levels and set V Sync Count to "Don't Sync") 

I seem to remember reading somewhere that it was recommended to turn V Sync off for VR (possibly OSVR specifically) anyway.

**Comment by DuFF14** [comment]
Correct, V-sync should be set to "Don't Sync"

**Comment by DuFF14** [comment]
@amiga any updates on this issue? are you using HDK2 or another HMD?

Regarding you original post, there shouldn't be any compatibility issues with OSVR-Core and OSVR-Unity. If Rendermanager fails to create, it's more likely a driver, hardware, firmware, or configuration issue. 

Not sure if this helps but FWIW, my workflow is to install the latest OSVR SDK from [here](http://access.osvr.com/binary/osvr-sdk-installer), and then download the latest Unity package from [here](http://access.osvr.com/binary/osvr-unity), replacing the one in my existing Unity project by closing the project, opening it again, and importing the package before hitting play. If RM fails to create in Unity, I run the RenderManager example programs included with the SDK to see if it's just a Unity issue or not. Usually if there's an issue, the RM example programs won't work either, then it's a matter of troubleshooting the issues mentioned above. If the RM examples do work and RM still fails in Unity, I test a fresh import of the plugin in a new project. 

Definitely make sure your firmware is updated if you are using HDK, there are some known issues with direct mode with older firmware.

**Comment by amiga** [comment]
HDK2 with newest firmware (hdk2svr-2.00.hex) I am still getting it in good quality or above, I am sticking to "fastest" to avoid it. To be honest I have not tried turning off vsync yet, I need to give that a go.

**Comment by DuFF14** [comment]
Confirmed that vsync causes a freeze.
